### PR TITLE
update langserver version to 0.34

### PR DESCRIPTION
**Pull request by appelgriebsch**
update the jdt language server to the latest release version 0.34.0

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -11,9 +11,9 @@ const {
   removeUnusedImportProvider
 } = actionProviders
 
-const serverDownloadUrl = 'http://download.eclipse.org/jdtls/milestones/0.14.0/jdt-language-server-0.14.0-201802282111.tar.gz'
+const serverDownloadUrl = 'http://download.eclipse.org/jdtls/milestones/0.34.0/jdt-language-server-0.34.0-201902280418.tar.gz'
 const serverDownloadSize = 35873467
```
Comment (50Wliu) [inline lib/main.js]: I believe this also needs to changed, right?

**Comment by 50Wliu** [comment]
We also start to hit https://github.com/eclipse/eclipse.jdt.ls/issues/651 after this upgrade.

File: lib/main.js (hunk 1 above)
Comment (appelgriebsch) [reply lib/main.js]: Not sure if this is checked at all. I just changed these two lines in my local copy, restarted Atom, and it downloads and uses the new version. Eventually the download size setting is only used for the busy-signal notification for the download status...

Comment (50Wliu) [reply lib/main.js]: It's not checked but it is used and resulted in a downlod percentage of 106% for me...

Comment (appelgriebsch) [reply lib/main.js]: ok, updated the size accordingly.

**Comment by 50Wliu** [comment]
Okay, I'm getting the following internal compiler error:
internal-compiler-error

I'll need to investigate some more about why this might be happening.

**Comment by 50Wliu** [comment]
Bug report: https://bugs.eclipse.org/bugs/show_bug.cgi?id=540738

I also get the following `window/logMessage` diagnostic info:
```
31 mar. 2019 3:52:55 
null
org.eclipse.jface.text.BadLocationException
	at org.eclipse.jface.text.ListLineTracker.getLineOffset(ListLineTracker.java:197)
	at org.eclipse.jface.text.AbstractLineTracker.getLineOffset(AbstractLineTracker.java:160)
	at org.eclipse.jface.text.AbstractDocument.getLineOffset(AbstractDocument.java:876)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.convertRange(WorkspaceDiagnosticsHandler.java:335)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.toDiagnostic(WorkspaceDiagnosticsHandler.java:315)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.lambda$3(WorkspaceDiagnosticsHandler.java:300)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.toDiagnosticsArray(WorkspaceDiagnosticsHandler.java:302)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.publishDiagnostics(WorkspaceDiagnosticsHandler.java:253)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandler.publishDiagnostics(WorkspaceDiagnosticsHandler.java:188)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$0(JDTLanguageServer.java:228)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$30(JDTLanguageServer.java:821)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**Pull request closed without merging**